src/site/topic16.rst

```diff
@@ -134,6 +134,19 @@ Array Iterator
     * If we want to iterate over the collection again, we create a new iterator
 
 
+.. warning::
+
+    Although in this example our iterator goes in our defined direction, there is nothing stopping us from creating an
+    iterator that goes in the reverse order.
+
+
+.. warning::
+
+    We need to be careful about modifying the collection when using an iterator. For example, with the array iterator,
+    the iterator has reference to the array that is being used and referenced in the thing that we want to iterate over.
+    If we were to modify something with our iterator, it would impact the thing we are iterating over.
```

Comment:
Not technically true. [Iterator](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Iterator.html) even has `remove` and it's definitely reasonable to update internal states depending on the use case (eg, collections in `java.util` keep track of a modification count to help prevent concurrent writes).

If you prefix this whole thing with _generally_, then I think it applies.